python/equistore-operations/equistore/operations/_dispatch.py

```diff
@@ -112,6 +112,20 @@ def copy(array):
         raise TypeError(UNKNOWN_ARRAY_TYPE)
 
 
+def eye_like(array, size: int):
+    """
+    Create an identity matrix with the given ``size``, and the same
+    dtype and device as ``array``.
+    """
+
+    if isinstance(array, TorchTensor):
+        return torch.eye(size).to(array.dtype).to(array.device)
```

Comment:
```suggestion
        return torch.eye(size).to(dtype=array.dtype, device=array.device)
```

Reply:
Unfortunately this single call doesn't map to the `aten` functions and it's necessary to split it into a call that changes the dtype and one that changes the device

Reply:
ok!